### PR TITLE
refactor: Migrate SeekBar to React

### PR DESCRIPTION
**Pull request by bennycode**

### CODE REVIEW

File: src/script/components/asset/controls/SeekBar.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      }}
+      onMouseUp={() => {
+        mediaElement.play();
+        setIsSeekBarThumbDragged(false);
+      }}
+      onMouseEnter={() => setIsSeekBarMouseOver(true)}
+      onMouseLeave={() => setIsSeekBarMouseOver(false)}
+      style={
+        {
+          '--seek-bar-progress': `${progress.toString(10)}%`,
```
Comment (ffflorian) [inline src/script/components/asset/controls/SeekBar.tsx:94]: Does this work?

```suggestion
          seekBarProgress: `${progress.toString(10)}%`,
```

File: src/script/components/asset/controls/SeekBar.test.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import {act} from '@testing-library/react';
```
Comment (ffflorian) [inline src/script/components/asset/controls/SeekBar.test.tsx:20]: Missing copyright header

**Review by ffflorian — CHANGES_REQUESTED** [review]

File: src/script/components/asset/controls/SeekBar.test.tsx (hunk 2 above)
Comment (bennycode) [reply src/script/components/asset/controls/SeekBar.test.tsx:20]: Thanks for the hint!

File: src/script/components/asset/controls/SeekBar.tsx (hunk 1 above)
Comment (bennycode) [reply src/script/components/asset/controls/SeekBar.tsx:94]: Nope, it's a custom property.

**Pull request merged**